doc/Operation.md

```diff
@@ -0,0 +1,328 @@
+
+The next code demonstrates the use of `Poll()`.
+
+```cpp
+/**
+ * Consuming Operation<T> interface to Poll for LRO updates.
+ *
+ */
+auto operationResult = client.ApiReturnOperation(...);
+
+while (true)
+{
+    // Get a read-only ref to the Http raw response after polling. Poll will update the status.
+    auto& response = operationResult.Poll();
+
+    if(operationResult.IsDone()) {
+        // Operation completed.
+        break;
+    }
+
+    // Read the http headers in the response
+    for (auto& header : response.GetHeaders()) {
+        // ... consumer-code
+    }
+}
+
+// Check is the Operation has created the value `T`
+if (!operationResult.HasValue())
+{
+    // The Operation completed but it couldn't build the value `T`. Most likely the operation stage ended with error or it was cancelled.
+    throw std::runtime_error("Operation finished with status:" + operationResult.Status());
+}
+
+// At this point, the Operation is success and we can get its value.
+auto valueT = operationResult.Value();
+```
+
+### PollUntilDone
+
+The `PollUntilDone()` is a convenience method for repeatedly call `Poll()` until the status is updated to a _final stage_. A `Response<T>` will be returned wrapping the model `T` and the Http raw response from which it was created. This method is useful when we are not interested about the Http raw response for each polling request.
+
+The `PollUntilDone()` method requires an `std::chrono::milliseconds` input parameter which controls how much time to wait between calling `Poll`.
+
+In the code below, instead of manually polling for updates, the `PollUntilDone` method is called.
+
+```cpp
+/**
+ * Consuming Operation<T> interface. Calling PollUntilDone.
+ *
+ */
+auto operationResult = client.ApiReturnOperation(...);
+
+try
+{
+    // Update each second
+    auto response = operationResult.PollUntilDone(std::chrono::milliseconds(1000));
+
+    // No need to check for `HasValue` here. The Operation will throw if the value can't be created after completing.
+    auto valueT = response.ExtractValue();
+    auto rawResponse = response.ExtractRawResponse();
+
+}
+catch (Azure::Core::RequestFailedException const& e)
+{
+    // Analyze the exception or re-throw.
+}
+```
+
+## Operation\<T> lifetime
+
+The `Operation<T>` holds an Http raw response. Each Operation\<T> derived class defines the moment when the Http raw response is moved to the Operation. This means that an Operation can be created without initially holding an Http raw response. In this case, calling `GetRawResponse` method from the Operation will `throw` as no http raw response has been set for it yet.
```

Comment:
We're probably getting rid of the raw response.

Reply:
We need to own an http raw response if we ever want to return a `Response<T>`.